package sms.entities.protobuf;

option java_package = "org.aaron.sms.entities.protobuf";
option java_outer_classname = "SMSProtocol";

message ClientToServerMessage {

  enum ClientToServerMessageType {
    CLIENT_SUBSCRIBE_TO_TOPIC = 0;
    CLIENT_UNSUBSCRIBE_FROM_TOPIC = 1;
    CLIENT_SEND_MESSAGE_TO_TOPIC = 2;
  }
  
  required ClientToServerMessageType messageType = 1;
  
  required string topicName = 2;
  
  optional bytes messagePayload = 3;
  
}

message ServerToClientMessage {

  enum ServerToClientMessageType {
    SERVER_TOPIC_MESSAGE_PUBLISH = 0;
  }
  
  required ServerToClientMessageType messageType = 1;
  
  required string topicName = 2;
  
  optional bytes messagePayload = 3;

}